src/main/kotlin/it/pagopa/wallet/eventdispatcher/configuration/WalletCDCClientConfiguration.kt

```diff
@@ -20,18 +20,21 @@ class WalletCDCClientConfiguration {
         walletCDCConfiguration: WalletCDCConfiguration
     ): ReactiveKafkaProducerTemplate<String, Any> {
         val configProps =
-            mapOf(
+            mutableMapOf<String, Any>(
                 ProducerConfig.BOOTSTRAP_SERVERS_CONFIG to walletCDCConfiguration.bootstrapServers,
                 ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG to StringSerializer::class.java,
                 ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG to JsonSerializer::class.java,
                 ErrorHandlingDeserializer.VALUE_DESERIALIZER_CLASS to
-                    ErrorHandlingDeserializer::class.java,
-                SaslConfigs.SASL_JAAS_CONFIG to
-                    "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"\$ConnectionString\" password=\"${walletCDCConfiguration.connectionString}\";",
-                SaslConfigs.SASL_MECHANISM to "PLAIN",
-                CommonClientConfigs.SECURITY_PROTOCOL_CONFIG to "SASL_SSL"
+                    ErrorHandlingDeserializer::class.java
             )
 
+        // Local testing in Docker do not require SASL
+        if (!walletCDCConfiguration.connectionString.startsWith("kafka")) {
```

Comment:
Can we handle this with spring profiles? Let's inject spring.profiles.active=local for local env and use a specific configuration for it

Reply:
done here 272f24f82d93e2db9f2005075d68291b206a409e